packages/amplify-category-auth/resources/cloudformation-templates/auth-template-force-alias.yml.ejs

```diff
@@ -0,0 +1,1240 @@
+      Principal: "cognito-idp.amazonaws.com"
+      FunctionName: !Ref function<%=props.resourceName%>PostAuthenticationName
+      SourceArn: !GetAtt UserPool.Arn
+  <% } %>
+  <%if (props.dependsOn.find(i => i.resourceName.includes('PostConfirmation'))) { %>
+  UserPoolPostConfirmationLambdaInvokePermission:
+    Type: "AWS::Lambda::Permission"
+    DependsOn: UserPool
+    Properties:
+      Action: "lambda:invokeFunction"
+      Principal: "cognito-idp.amazonaws.com"
+      FunctionName: !Ref function<%=props.resourceName%>PostConfirmationName
+      SourceArn: !GetAtt UserPool.Arn
+  <% } %>
+  <%if (props.dependsOn.find(i => i.resourceName.includes('PreAuthentication'))) { %>
+  UserPoolPreAuthenticationLambdaInvokePermission:
+    Type: "AWS::Lambda::Permission"
+    DependsOn: UserPool
+    Properties:
+      Action: "lambda:invokeFunction"
+      Principal: "cognito-idp.amazonaws.com"
+      FunctionName: !Ref function<%=props.resourceName%>PreAuthenticationName
+      SourceArn: !GetAtt UserPool.Arn
+  <% } %>
+  <%if (props.dependsOn.find(i => i.resourceName.includes('PreSignup'))) { %>
+  UserPoolPreSignupLambdaInvokePermission:
+    Type: "AWS::Lambda::Permission"
+    DependsOn: UserPool
+    Properties:
+      Action: "lambda:invokeFunction"
+      Principal: "cognito-idp.amazonaws.com"
+      FunctionName: !Ref function<%=props.resourceName%>PreSignupName
+      SourceArn: !GetAtt UserPool.Arn
+  <% } %>
+  <%if (props.dependsOn.find(i => i.resourceName.includes('PreTokenGeneration'))) { %>
+  UserPoolPreTokenGenerationLambdaInvokePermission:
+    Type: "AWS::Lambda::Permission"
+    DependsOn: UserPool
+    Properties:
+      Action: "lambda:invokeFunction"
+      Principal: "cognito-idp.amazonaws.com"
+      FunctionName: !Ref function<%=props.resourceName%>PreTokenGenerationName
+      SourceArn: !GetAtt UserPool.Arn
+  <% } %>
+  <%if (props.dependsOn.find(i => i.resourceName.includes('VerifyAuthChallengeResponse'))) { %>
+  UserPoolVerifyAuthChallengeResponseLambdaInvokePermission:
+    Type: "AWS::Lambda::Permission"
+    DependsOn: UserPool
+    Properties:
+      Action: "lambda:invokeFunction"
+      Principal: "cognito-idp.amazonaws.com"
+      FunctionName: !Ref function<%=props.resourceName%>VerifyAuthChallengeResponseName
+      SourceArn: !GetAtt UserPool.Arn
+  <% } %>
+  # Updating lambda role with permissions to Cognito
+  <% if (props.permissions && props.permissions.length > 0) { %>
+  <% for(var i=0; i < props.permissions.length; i++) { %>
+  <%=`${props.resourceName}${JSON.parse(props.permissions[i]).trigger}${JSON.parse(props.permissions[i]).policyName}:`%>
+    Type: AWS::IAM::Policy
+    Properties:
+      PolicyName: <%=`${props.resourceName}${JSON.parse(props.permissions[i]).trigger}${JSON.parse(props.permissions[i]).policyName}`%>
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: <%= JSON.parse(props.permissions[i]).effect %>
```

Comment:
Instead of JSON.parse, do we need a safe JSON.parse here ? in case the parameter is not valid JSON  ?

Reply:
I think it is actually better / less confusing to crash if the props are not valid json vs invisibly ignoring or whatever other safety mechanism